Initial commit.

PiekosEmil opened the pull request.

PiekosEmil committed:
Initial commit.

marcinkunert commented on src/main/java/pl/javastart/task/Room.java:
```diff
@@ -0,0 +1,21 @@
+        this.acMounted = acMounted;
+    }
+
+    void coolDown() {
+        if (acMounted && (temperature > minTemperature)) {
+            temperature--;
+        }
+    }
```
1. Metoda miała zwracać informację czy udało się obniżyć temperaturę:

> . Metoda powinna zwracać true, jeżeli udało się obniżyć temperaturę, lub false jeżeli w pokoju nie ma klimatyzatora lub osiągnięto temperaturę graniczną.

2. Jeśli temperatura to 20,5*C, a minimalna to 20*C, to po uruchomieniu metody będzie 19,5*C, czyli mniej niż minimalna.

PiekosEmil committed:
Poprawiony kod.

Obecnie metoda coolDown obniża temperaturę tylko do określonego poziomu.
Jednak metoda coolDownIsPossible, która ma zwracać true jeśli udało się obniżyc temperaturę nie zwraca wartości true jeśli nie osiągnie temperatury minimalnej.

Próbowałem różnych możliwości, żeby wyświetlany rezultat zgadzał się z logiką działania klimatyzacji. Czyli jeśli udało się obniżyć temperaturę z 23.7 do 19.7 to powinna wyświetlić się wartość true jednak. Jednak nie mam pojęcia jak uzależnić aktualną temperaturę od temperatury minimalnej. Aktualnie metoda coolDownIsPossible zwraca true tylko wtedy kiedy osiągnięta jest temperatura minimalna. Nie wiem jak uzależnić wyświetlanie informacji o obniżeniu temperatury od różnicy temperatury początkowej i końcowej.

slawekludwiczak commented on src/main/java/pl/javastart/task/Room.java:
```diff
@@ -0,0 +1,45 @@
+
+    void coolDown() {
+        if (acMounted) {
+            if (temperature > minTemperature) {
```
ten if nie jest potrzebny

slawekludwiczak commented on src/main/java/pl/javastart/task/Room.java:
```diff
@@ -0,0 +1,45 @@
+        }
+    }
+
+    boolean coolDownIsPossible() {
+        if (!(acMounted && temperature > minTemperature || temperature != minTemperature)) {
+            return true;
+        } else if (acMounted && temperature > minTemperature) {
+            if (temperature < minTemperature) {
+                return true;
+            } else {
+                return false;
+            }
+        }
+        return false;
+    }
```
W zadaniu nie ma mowy o metodzie, która zwraca true/false, ale nic poza tym nie robi, powinieneś połączyć metody `coolDown` i `coolDownIsPossible` w jedną

slawekludwiczak reviewed (changes requested):
code review https://javastartbootcamp.slack.com/archives/C072BTXL6PM/p1716218223895739

PiekosEmil committed:
Skompresowałem napisane metody do jednej.
Wprowadziłem zupełnie nowe warunki logiczne.
Wyświetlanie w konsoli jest zgodne ze stanem faktycznym.
Dodano komentarze opisujące działanie lub niedziałanie metody coolDown.

slawekludwiczak commented on src/main/java/pl/javastart/task/Room.java:
```diff
@@ -0,0 +1,27 @@
+    boolean coolDown() {
+        if (acMounted && temperature - 1 > minTemperature) {
+            temperature--;
+            return temperature - 1 < minTemperature;
```
tutaj można po prostu zwrócić true

slawekludwiczak commented on src/main/java/pl/javastart/task/Room.java:
```diff
@@ -0,0 +1,27 @@
+            return temperature - 1 < minTemperature;
+        } else if (acMounted && temperature > minTemperature) {
+            temperature = temperature - (temperature - minTemperature);
+            return temperature == minTemperature;
```
tutaj można po prostu zwrócić true

slawekludwiczak commented on src/main/java/pl/javastart/task/Room.java:
```diff
@@ -0,0 +1,27 @@
+            temperature--;
+            return temperature - 1 < minTemperature;
+        } else if (acMounted && temperature > minTemperature) {
+            temperature = temperature - (temperature - minTemperature);
```
to jest to samo co `temperature = minTemperature`

slawekludwiczak commented on src/main/java/pl/javastart/task/Room.java:
```diff
@@ -0,0 +1,27 @@
+            temperature = temperature - (temperature - minTemperature);
+            return temperature == minTemperature;
+        }
+        return acMounted && temperature == minTemperature;
```
Tutaj nie obniżamy temperatury, więc można po prostu zwrócić false

slawekludwiczak reviewed (changes requested):
code review https://javastartbootcamp.slack.com/archives/C072BTXL6PM/p1716275565606569
